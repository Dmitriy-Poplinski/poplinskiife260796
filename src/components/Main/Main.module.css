.main {
    height: 88%;
    display: flex;
};
