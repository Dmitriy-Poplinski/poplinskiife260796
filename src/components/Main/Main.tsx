import React from 'react';

import { LeftPanel } from '../LeftPanel/LeftPanel';
import { MainBody } from '../MainBody/MainBody';
import { RightPanel } from '../RightPanel/RightPanel';

import classes from './Main.module.css';

interface MainProps {
    images: string[],
    shuffleImageArray: () => void
}

export function Main({ images, shuffleImageArray }: MainProps) {
    const [ left_panel_image, main_body_image, right_panel_image ] = images;

    return (
        <div className={classes.main} data-testid='main'>
            <LeftPanel image={left_panel_image}/>
            <MainBody image={main_body_image} shuffleImageArray={shuffleImageArray}/>
            <RightPanel image={right_panel_image}/>
        </div>
    );
}
