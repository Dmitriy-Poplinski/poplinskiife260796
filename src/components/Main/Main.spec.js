import { render, screen } from '@testing-library/react';

import { Main } from './Main';

describe('Main:', () => {

    let data; 

    test('Main gets array', () => {
        data = ['testing_image', 'testing_image_2', 'testing_image_3'];

        render(<Main images={data} />);

        expect(screen.getByTestId('main')).toMatchSnapshot();
    });
});
