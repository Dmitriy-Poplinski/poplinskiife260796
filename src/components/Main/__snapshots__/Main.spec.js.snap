// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main: Main gets array 1`] = `
<div
  class="main"
  data-testid="main"
>
  <div
    class="left_panel"
    data-testid="left_panel"
    style="background: orange;"
  >
    Left Panel
  </div>
  <div
    class="main_body"
    data-testid="main_body"
    style="background: violet;"
  >
    <button
      class="button"
      data-testid="button"
    >
      Click me
    </button>
  </div>
  <div
    class="right_panel"
    data-testid="right_panel"
    style="background: maroon;"
  >
    Right Panel
  </div>
</div>
`;
